### PR TITLE
fix(jobs): upload an index file if it does not exist

### PR DESCRIPTION
**Pull request by bacongobbler**
When creating a release for the first time, an index file is not present in the bucket to copy and merge from. Therefore in this case we just want to upload an initial index file and call it a day.

Note that this will not help jenkins-node, but will prevent this issue from occurring in the future.

### CODE REVIEW

File: jobs/component_chart_publish.groovy
Comment (vdice) [inline jobs/component_chart_publish.groovy]: we need to escape the `$` here (`\$`) b/c it is a shell/runtime resolution as opposed to a groovy/job build-time resolution...

**Review by vdice — CHANGES_REQUESTED** [review]

**Comment by vdice** [comment]
minor change needed; otherwise LGTM!

**Comment by bacongobbler** [comment]
minor change addressed

**Review by vdice — APPROVED** [review]

**Review by kmala — APPROVED** [review]

**Comment by bacongobbler** [comment]
The code has been changed to accomodate if the file does not exist, please take a look before this is merged.

The code was thanks to http://stackoverflow.com/a/27003199.

**Comment by bacongobbler** [comment]
Hold up on this PR. Just a thought, since this job runs when no index file exists, should we run `helm package` initially, upload the index file then re-run `helm package` and merge it with the now-present index? Just wondering if uploading a bare index file when we're releasing a tag of a chart is what we intend.

File: jobs/component_chart_publish.groovy
Comment (vdice) [inline jobs/component_chart_publish.groovy]: could also use `if [ -z "\$(aws s3 ls s3://helm-charts/${repo_chart}/index.yaml)" ]; then` here -- but just an fyi

**Comment by vdice** [comment]
> should we run `helm package` initially

@bacongobbler we do a helm package immediately prior, so the first `index.yaml` file _should_ actually be populated... so I think we are good here.  But good concern.

File: jobs/component_chart_publish.groovy
Comment (bacongobbler) [reply jobs/component_chart_publish.groovy]: done

File: jobs/component_chart_publish.groovy
Comment (vdice) [inline jobs/component_chart_publish.groovy]: ah, also `s/repo_chart/repo.chart`

**Pull request merged**
